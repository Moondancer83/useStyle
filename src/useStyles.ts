import { makeStyles } from "@material-ui/styles";
// eslint-disable-next-line no-unused-vars
import { PropsOfStyles, Styles, WithStylesOptions } from "@material-ui/styles/withStyles";
// eslint-disable-next-line no-unused-vars
import { DefaultTheme } from "@material-ui/styles/defaultTheme";
// eslint-disable-next-line no-unused-vars
import { Omit } from "@material-ui/types";
// eslint-disable-next-line no-unused-vars
import { StylesHook } from "@material-ui/styles/makeStyles/makeStyles";

export function useStyles<S, Theme = DefaultTheme, Props extends {} = {}, ClassKey extends string = string>(
  styles: Styles<Theme, Props, ClassKey>,
  options?: Omit<WithStylesOptions<Theme>, "withTheme">,
  props?: any | PropsOfStyles<S>
): Record<ClassKey, string> {
  const hook: StylesHook<Styles<Theme, Props, ClassKey>> = makeStyles(styles, options);
  return hook(props);
}

export function useNamedStyles<S, Theme = DefaultTheme, Props extends {} = {}, ClassKey extends string = string>(
  classNamePrefix: string,
  styles: Styles<Theme, Props, ClassKey>,
  options?: Omit<WithStylesOptions<Theme>, "withTheme">,
  props?: any
) {
  return useStyles(styles, { ...options, classNamePrefix }, props);
}
