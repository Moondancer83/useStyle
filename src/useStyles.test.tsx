import React from "react";
import renderer from 'react-test-renderer';
import {useStyles, useNamedStyles} from "./useStyles";

function TestComponent() {
  let classes = useStyles({root: {color: "red"}});

  return <div className={classes.root}/>
}

test("useStyles returns classes object", () => {
  const component = renderer.create(
    <TestComponent />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  // @ts-ignore
  expect(tree.props.className).toBe("makeStyles-root-1");
});



function Test2Component() {
  let classes = useNamedStyles("Test2", {root: {color: "red"}});

  return <div className={classes.root}/>
}

test("useNamedStyles returns classes object", () => {
  const component = renderer.create(
    <Test2Component />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  // @ts-ignore
  expect(tree.props.className).toBe("Test2-root-2");
});
