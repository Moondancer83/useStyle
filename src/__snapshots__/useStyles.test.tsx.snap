// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useNamedStyles returns classes object 1`] = `
<div
  className="Test2-root-2"
/>
`;

exports[`useStyles returns classes object 1`] = `
<div
  className="makeStyles-root-1"
/>
`;
